test/move_effect_sticky_web.c

```diff
@@ -0,0 +1,237 @@
+        MESSAGE("Foe Shuckle's Speed rose!");
+    }
+}
+
+#define BATTLER_OPPONENT (opponentSetUpper == 0 ? opponentLeft : opponentRight)
+#define BATTLER_PLAYER (playerSetUpper == 0 ? playerLeft : playerRight)
+
+DOUBLE_BATTLE_TEST("Sticky Web has correct interactions with Mirror Armor - the battler which set up Sticky Web has its Speed lowered instead")
+{
+    u8 playerSetUpper, opponentSetUpper; // 0 left, 1 right
+
+    PARAMETRIZE {playerSetUpper = 0; opponentSetUpper = 0;}
+    PARAMETRIZE {playerSetUpper = 0; opponentSetUpper = 1;}
+    PARAMETRIZE {playerSetUpper = 1; opponentSetUpper = 0;}
+    PARAMETRIZE {playerSetUpper = 1; opponentSetUpper = 1;}
+
+    GIVEN {
+        ASSUME(P_GEN_8_POKEMON == TRUE);
+        PLAYER(SPECIES_SQUIRTLE);
+        PLAYER(SPECIES_CHARMANDER);
+        PLAYER(SPECIES_CORVIKNIGHT) {Ability(ABILITY_MIRROR_ARMOR); Item(ITEM_IRON_BALL); } // Iron Ball, so that flying type Corviknight is affected by Sticky Web.
+        OPPONENT(SPECIES_CATERPIE);
+        OPPONENT(SPECIES_WEEDLE);
+    } WHEN {
+        TURN { MOVE(BATTLER_OPPONENT, MOVE_STICKY_WEB); }
+        TURN { MOVE(BATTLER_PLAYER, MOVE_STICKY_WEB); }
+        TURN { SWITCH(playerRight, 2); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_STICKY_WEB, BATTLER_OPPONENT);
+        MESSAGE("A sticky web spreads out on the ground around your team!");
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_STICKY_WEB, BATTLER_PLAYER);
+        MESSAGE("A sticky web spreads out on the ground around the opposing team!");
+
+        MESSAGE("Go! Corviknigh!");
+        MESSAGE("Corviknigh was caught in a Sticky Web!");
+        ABILITY_POPUP(playerRight, ABILITY_MIRROR_ARMOR);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, BATTLER_OPPONENT);
+        if (opponentSetUpper == 0) {
+            MESSAGE("Foe Caterpie's Speed fell!");
+        } else {
+            MESSAGE("Foe Weedle's Speed fell!");
+        }
+    }
+}
+
+#undef BATTLER_OPPONENT
+#undef BATTLER_PLAYER
+
+DOUBLE_BATTLE_TEST("Sticky Web has correct interactions with Mirror Armor - no one has their Speed lowered if the set upper switched")
+{
+    u16 speedPlayer, speedOpponent;
+
+    // We need to make sure Sticky Web user saves for both sides, so it doesn't matter who sets it first.
+    PARAMETRIZE {speedPlayer = 5; speedOpponent = 10;}
+    PARAMETRIZE {speedPlayer = 10; speedOpponent = 5;}
+
+    GIVEN {
+        ASSUME(P_GEN_8_POKEMON == TRUE);
+        PLAYER(SPECIES_SQUIRTLE) {Speed(speedPlayer); }
+        PLAYER(SPECIES_CHARMANDER) {Speed(speedPlayer); }
+        PLAYER(SPECIES_CORVIKNIGHT) {Ability(ABILITY_MIRROR_ARMOR); Item(ITEM_IRON_BALL); Speed(speedOpponent); } // Iron Ball, so that flying type Corviknight is affected by Sticky Web.
+        OPPONENT(SPECIES_CATERPIE) {Speed(speedOpponent); }
+        OPPONENT(SPECIES_WEEDLE) {Speed(speedOpponent); }
+        OPPONENT(SPECIES_PIDGEY) {Speed(speedOpponent); } // Flying type,so not affected by Sticky Web.
```

Comment:
```suggestion
        ASSUME(P_GEN_8_POKEMON == TRUE);
        PLAYER(SPECIES_SQUIRTLE) { Speed(speedPlayer); }
        PLAYER(SPECIES_CHARMANDER) { Speed(speedPlayer); }
        PLAYER(SPECIES_CORVIKNIGHT) { Ability(ABILITY_MIRROR_ARMOR); Item(ITEM_IRON_BALL); Speed(speedOpponent); } // Iron Ball, so that flying type Corviknight is affected by Sticky Web.
        OPPONENT(SPECIES_CATERPIE) { Speed(speedOpponent); }
        OPPONENT(SPECIES_WEEDLE) { Speed(speedOpponent); }
        OPPONENT(SPECIES_PIDGEY) { Speed(speedOpponent); } // Flying type,so not affected by Sticky Web.
```